raiden/network/proxies/user_deposit.py

```diff
@@ -23,14 +48,113 @@ def __init__(
             CONTRACT_USER_DEPOSIT,
         )
 
+        self.client = jsonrpc_client
+
         self.address = user_deposit_address
+        self.node_address = self.client.address
         self.contract_manager = contract_manager
+
         self.proxy = jsonrpc_client.new_contract_proxy(
             self.contract_manager.get_contract_abi(CONTRACT_USER_DEPOSIT),
             to_normalized_address(user_deposit_address),
         )
 
-        self.client = jsonrpc_client
+        self.deposit_lock = RLock()
+
+    def token_address(
+            self,
+            block_identifier: BlockSpecification,
+    ) -> Address:
+        return to_canonical_address(self.proxy.contract.functions.token().call(
+            block_identifier=block_identifier,
+        ))
+
+    def get_total_deposit(
+            self,
+            address: Address,
+            block_identifier: BlockSpecification,
+    ) -> TokenAmount:
+        return self.proxy.contract.functions.balances(address).call(
+            block_identifier=block_identifier,
+        )
+
+    def deposit(
+            self,
+            beneficiary: Address,
+            total_deposit: TokenAmount,
+            block_identifier: BlockSpecification,
+    ) -> None:
+        """ Deposit provided amount into the user-deposit contract
+        to the beneficiary's account. """
+
+        token_address = self.token_address(block_identifier)
+        token = Token(
```

Comment:
Please, add a description here regarding race conditions across different proxies and why it may not be a problem here.